src/uart/mod.rs

```diff
@@ -69,24 +112,57 @@ pub enum CharacterLength {
 /// The clock source to use for UART
 #[repr(u8)]
 pub enum ClockSources {
-    PLCK = 0,
+    PCLK = 0,
     IBRO = 2,
 }
 
 /// # Stop Bits
 /// The number of stop bits to use.
 /// Note: When using a character length of five bits, passing the variant
 /// `TwoBits` uses 1.5 bits.
-#[repr(u8)]
 pub enum StopBits {
-    OneBit = 0,
-    TwoBits = 1,
+    OneBit,
+    TwoBits,
+}
+
+/// # Threshold Size
+/// Then number of bits to apply to a FIFO threshold
+#[repr(u8)]
+pub enum ThresholdSize {
+    Threshold1 = 1,
+    Threshold2 = 2,
+    Threshold3 = 3,
+    Threshold4 = 4,
+    Threshold5 = 5,
+    Threshold6 = 6,
+    Threshold7 = 7,
+    Threshold8 = 8,
+}
```

Comment:
This does not make sense as an enum, it would be better just to `debug_assert!` that the value as type `u8` falls within this range. 

Mostly, enums are better used for input that does not particularly have a nice numeric value associated to the caller (ie. the caller wouldn't expect to use a number to represent this input).